.github/workflows/matomo-tests.yml

```diff
@@ -0,0 +1,108 @@
+name: Matomo Tests
+
+on:
+  pull_request:
```

Comment:
I don't think, that our tests should in the future only run for pull requests. They should also run for every commit on any base branch (4.x-dev, 5.x-dev) and maybe even manually.
Allowing that will make some definitions below incorrect, as there will not always be a reference like `${{ github.event.pull_request.head.sha }}`